include/LightGBM/utils/log.h

```diff
@@ -14,6 +14,13 @@
 #include <iostream>
 #include <stdexcept>
 
+#ifdef LGB_R_BUILD
+  #define R_NO_REMAP
+  #define R_USE_C99_IN_CXX
+  #include <R_ext/Error.h>
+  #include <R_ext/Print.h>
+#endif
+
```

Comment:
for those who are new  to it:

* `R_NO_REMAP`:  Do not change symbols like `Rf_error()` to `error()`
* `R_USE_C99_IN_CXX`: Make  sure that `Rvprintf()` is defined. ([reference](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Printing))

Reply:
Note: `R_USE_C99_IN_CXX` is defined in `R_ext/Print.h` in R 4.0. Will this break when using R 4.0?